tests/test_prediction/test_target_mean_regressor.py

```diff
@@ -0,0 +1,172 @@
+]
+
+
+@pytest.mark.parametrize("_bins, _strategy", _false_input_params)
+def test_raises_error_when_wrong_input_params(_bins, _strategy):
+    with pytest.raises(ValueError):
+        assert TargetMeanRegressor(bins=_bins)
+    with pytest.raises(ValueError):
+        assert TargetMeanRegressor(strategy=_strategy)
+
+
+def test_default_params():
+    transformer = TargetMeanRegressor()
+    assert isinstance(transformer, TargetMeanRegressor)
+    assert transformer.variables is None
+    assert transformer.bins == 5
+    assert transformer.strategy == "equal_width"
+
+
+def test_attributes_upon_fitting(df_pred):
+    # Case 1: check all init params and class attributes
+    transformer = TargetMeanRegressor(
+        variables=None,
+        bins=5,
+        strategy="equal_width"
+    )
+    transformer.fit(df_pred[["City", "Age", "Studies"]], df_pred["Marks"])
+
+    # test init params
+    assert transformer.variables is None
+    assert transformer.bins == 5
+    assert transformer.strategy == "equal_width"
+    # test attributes
+    assert transformer.variables_categorical_ == ["City", "Studies"]
+    assert transformer.variables_numerical_ == ["Age"]
+    assert type(transformer.pipeline_
+                .named_steps["discretiser"]) == EqualWidthDiscretiser
+    assert type(transformer.pipeline_
+                .named_steps["encoder_num"]) == MeanEncoder
+    assert type(transformer.pipeline_
+                .named_steps["encoder_cat"]) == MeanEncoder
+    assert transformer.n_features_in_ == 3
+
+
+def test_target_mean_predictor_transformation(df_pred, df_pred_small):
+    # Case 2: Check transformation
+    transformer = TargetMeanRegressor(
+        variables=None,
+        bins=5,
+        strategy="equal_width"
+    )
+
+    transformer.fit(df_pred[["City", "Age"]], df_pred["Marks"])
+    predictions = transformer.predict(df_pred_small[["City", "Age"]]).round(6)
+
+    # test results
+    assert (predictions == pd.Series(
+        [0.483333, 0.583333, 0.391667, 0.666667, 0.3, 0.391667]
+    )).all()
+
+
+def test_regression_score_calculation_with_equal_frequency(
+        df_pred, df_pred_small
+):
+    # Case 3: check score() method
+    transformer = TargetMeanRegressor(
+        variables=None,
+        bins=5,
+        strategy="equal_frequency"
+    )
+
+    transformer.fit(df_pred[["City", "Age"]], df_pred["Marks"])
+    r2 = transformer.score(
+        df_pred_small[["City", "Age"]],
+        df_pred_small["Marks"]
+    )
+
+    # test R-Squared calc
+    assert r2.round(6) == -0.022365
+
+
+def test_predictor_with_all_numerical_variables(df_pred, df_pred_small):
+    # Case 4: Check predictor when all variables are numerical
+    transformer = TargetMeanRegressor(
+        variables=None,
+        bins=3,
+        strategy="equal_width"
+    )
+
+    transformer.fit(df_pred[["Age", "Height_cm"]], df_pred["Marks"])
+    r2 = transformer.score(
+        df_pred_small[["Age", "Height_cm"]], df_pred_small["Marks"]
+    )
+
```

Comment:
could we add an assert for the output of predict?

can we also test the transformer by passing a df with 1 variable and a df with 3 or more variables?